src/assets/TokenBalancesController.ts

```diff
@@ -96,7 +96,12 @@ export class TokenBalancesController extends BaseController<TokenBalancesConfig,
     const newContractBalances: { [address: string]: typeof BN } = {};
     for (const i in tokens) {
       const { address } = tokens[i];
-      newContractBalances[address] = await assetsContract.getBalanceOf(address, selectedAddress);
+      try {
+        newContractBalances[address] = await assetsContract.getBalanceOf(address, selectedAddress);
```

Comment:
We probably need to add `tokens[i].balanceError = null` here or something similar. So if for some reason the error was a one off thing and gets resolved the next time, it will be cleared. Otherwise the error will persist forever